core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -312,11 +312,15 @@ private[spark] object JsonProtocol {
         // We only have 3 kind of internal accumulator types, so if it's not int or long, it must be
         // the blocks accumulator, whose type is `java.util.List[(BlockId, BlockStatus)]`
         case v =>
-          JArray(v.asInstanceOf[java.util.List[(BlockId, BlockStatus)]].asScala.toList.map {
-            case (id, status) =>
-              ("Block ID" -> id.toString) ~
-              ("Status" -> blockStatusToJson(status))
-          })
+          val blockAccumulator =
```

Comment:
I don't see why this would help. You add a wrapper, but, synchronizing your local access to it doesn't do anything because nothing else is synchronizing on it.

PS can the `List[(BlockId,BlockStatus)]` type just be part of the match predicate?